extensions/wikia/UserRenameTool/process/ProcessGlobal.class.php

```diff
@@ -0,0 +1,597 @@
+<?php
+
+namespace UserRenameTool\Process;
+
+use UserRenameTool\Tasks\MultiWikiRename;
+use Wikia\Tasks\Queues\PriorityQueue;
+
+class ProcessGlobal extends ProcessBase {
+
+	const MAX_EXECUTION_TIME = 3600; // 1h
+
+	/**
+	 * Runs the whole rename process, schedules background jobs/tasks if needed.
+	 *
+	 * @return bool True if the process succeeded
+	 */
+	public function run() {
+		$this->logInfo( "User rename process started" );
+		if ( !$this->setup() ) {
+			return false;
+		}
+
+		$status = false;
+		try {
+			$status = $this->renameUser();
+		} catch ( \Exception $e ) {
+			$this->logError( "Global rename failed", [
+				'errorMessage' => $e->getMessage(),
+				'errorLocation' => $e->getFile(),
+				'errorLine' => $e->getLine()
+			] );
+			$this->addError(
+				wfMessage( 'userrenametool-error-cannot-rename-unexpected' )->inContentLanguage()->text()
+			);
+		}
+
+		if ( !$status ) {
+			$this->logFailToStaff();
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Checks if the request provided to the constructor is valid.
+	 *
+	 * @return bool True if all prerequisites are met
+	 */
+	protected function setup() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)